src/renderer/components/database-diagram.jsx

```diff
@@ -0,0 +1,125 @@
+import React, { Component, PropTypes } from 'react';
+import joint from 'jointjs';
+
+require('./jointjs-diagram-table.js');
+require('./jointjs-diagram-table-cell.js');
+require('jointjs/dist/joint.min.css');
+
+
+export default class DatabaseDiagram extends Component {
+  static propTypes = {
+    tables: PropTypes.array,
+    columnsByTable: PropTypes.object,
+    links: PropTypes.object,
+  }
+
+  constructor(props) {
+    super(props);
+    this.graph = new joint.dia.Graph();
+  }
+
+  componentDidMount() {
+    const { tables, columnsByTable, links } = this.props;
+
+    this.paper = new joint.dia.Paper({
+      el: $(this.refs.diagram),
+      width: $(this.refs.diagram).parent().width(),
+      height: 600,
+      model: this.graph,
+      gridSize: 1,
+      restrictTranslate: true,
+    });
+
+    const tableShapes = [];
+    const tableCells = [];
+    const tableLinks = [];
+
+    try {
+      let currentTable;
+      let newTabCell;
+      let newLink;
+
+      /* Tables & views */
+      tables.map((table, index) => {
+        tableShapes.push(new joint.shapes.sqlectron.Table({
+          position: {
+            x: 100 + (index % 6) * 100,
+            y: 30 + (index % 4) * 100,
+          },
+          size: {
+            width: 120,
+            height: (columnsByTable[table].length + 1.5) * 20,
+          },
+          name: `${table}`,
+        }));
+        currentTable = tableShapes[index];
+
+        columnsByTable[table].map((column, idx) => {
+          newTabCell = new joint.shapes.sqlectron.TableCell({
+            position: {
+              x: (currentTable.position().x),
+              y: (currentTable.position().y + (idx + 1) * 20),
+            },
+            size: {
+              width: 100,
+              height: 20,
+            },
+            name: `${column.name}`,
+            tableName: `${table}`,
+          });
+          currentTable.embed(newTabCell);
+          tableCells.push(newTabCell);
+        });
+      });
+
+      /* Links */
+      let targetIndex;
+
+      tables.map((table, index) => {
+        currentTable = tableShapes[index];
+
+        links[table].map((target) => {
+          targetIndex = tables.findIndex((t) => t === target.linksTo);
+          if (targetIndex !== -1) {
+            newLink = new joint.dia.Link({
+              source: { id: currentTable.id },
+              target: { id: tableShapes[targetIndex].id },
+            });
+            newLink.attr({'.marker-target': { fill: 'yellow', d: 'M 10 0 L 0 5 L 10 10 z' }});
+            tableLinks.push(newLink);
+          }
+        });
+      });
+    } catch (e) {
+      console.log(e);
```

Comment:
In case a error occurs. Shouldn't it be displayed to the user? But if the console.log is really required. An approach that I would suggest is using a suffix in the log, such as `console.log('Error rendering diagram', err);`. Because in case an error comes as undefined because of some internal problem. At least we know is passing through this catch.